### DISCUSSION TITLE
how to use Dfreds removeEffect?

### DISCUSSION THREAD

**Discussion by Terrox** [DFreds/dfreds-convenient-effects#196] [General]
Why doesn't this macro work on a token item? 
on use it should create or remove the existing effect named TEST.
creates a new effect called TEST,  detects it has the effect called TEST but removeEffect fails to work, says cannot find effect named TEST. uuid is correct (I think).

```
let sourceActorData = args.length == 0 ? token.data.actorData : args[0].actor; 
let uuid = args.length ? args[0].actorUuid : token.actor.uuid;
let newEffectName = "TEST";
let hasEffect = game.dfreds.effectInterface.hasEffectApplied(newEffectName, uuid);
if (!hasEffect) {
    let effectData = game.dfreds.effectInterface.findEffectByName('Bane').convertToObject();
    effectData.name = newEffectName;
    game.dfreds.effectInterface.addEffectWith({ effectData, uuid });
} else{
console.log('trying to remove effect '+newEffectName+' from '+uuid );
  game.dfreds.effectInterface.removeEffect({ effectName: newEffectName, uuid });
}
```
[fvtt-Item-remove-effect-test-case.json.txt](https://github.com/DFreds/dfreds-convenient-effects/files/10200847/fvtt-Item-remove-effect-test-case.json.txt)

**Comment by Terrox** [comment]
Foundry V9.280, dfred 2.7.3

**Comment by DFreds** [comment]
@Terrox is it adding the effect correctly? Just not removing it?
